test/e2e/service.go

```diff
@@ -989,56 +1057,81 @@ spec:
+			// the lifecycle of the test.
 			// If `kindexgw` exists, connect client container to it
-			runCommand(containerRuntime, "network", "connect", "kindexgw", clientContainerName)
+			kindExGWNetwork, err := infraprovider.Get().GetNetwork("kindexgw")
+			if err == nil {
+				_, _ = providerCtx.AttachNetwork(kindExGWNetwork, serverExternalContainer.GetName())
+			}
 
 			ginkgo.By("Selecting additional IP addresses for each node")
 			// add new secondary IP from node subnet to all nodes, if the cluster is v6 add an ipv6 address
-			toCurlAddresses := sets.NewString()
+			secondaryNetworkV4Subnet, secondaryNetworkV6Subnet, err := secondaryProviderNetwork.GetIPv4IPv6Subnets()
+			framework.ExpectNoError(err, "must get secondary provider network subnets")
+			secondarySubnetStr := secondaryNetworkV4Subnet
+			if isIPv6Cluster {
+				secondarySubnetStr = secondaryNetworkV6Subnet
+			}
+			_, secondarySubnet, err := net.ParseCIDR(secondarySubnetStr)
+			framework.ExpectNoError(err, "must parse secondary subnet %q", secondarySubnetStr)
+			toCurlAddressesSecondary := sets.NewString()
 			for i, node := range nodes.Items {
-
 				addrAnnotation, ok := node.Annotations["k8s.ovn.org/host-cidrs"]
 				gomega.Expect(ok).To(gomega.BeTrue())
-
-				var addrs []string
-				err := json.Unmarshal([]byte(addrAnnotation), &addrs)
+				var allAddrs []string
+				var filteredAddrs []string
+				err := json.Unmarshal([]byte(addrAnnotation), &allAddrs)
 				framework.ExpectNoError(err, "failed to parse node[%s] host-address annotation[%s]", node.Name,
 					addrAnnotation)
-				for i, addr := range addrs {
+				for _, addr := range allAddrs {
 					addrSplit := strings.Split(addr, "/")
 					gomega.Expect(addrSplit).Should(gomega.HaveLen(2))
-					addrs[i] = addrSplit[0]
+					// ensure only IPs for the secondary subnet are added
+					ip := net.ParseIP(addrSplit[0])
+					gomega.Expect(ip).NotTo(gomega.BeNil())
+					if !secondarySubnet.Contains(ip) {
+						framework.Logf("skipping IP %s because it is not within networks subnet %s", ip, subnet)
+						continue // skip if not within the networks subnet
+					}
+					filteredAddrs = append(filteredAddrs, ip.String())
 				}
-				toCurlAddresses.Insert(addrs...)
+				toCurlAddressesSecondary.Insert(filteredAddrs...)
 
 				// Calculate and store for AfterEach new target IP addresses.
-				var newIP string
 				if nodeIPs[node.Name] == nil {
 					nodeIPs[node.Name] = make(map[int]string)
 				}
-				if utilnet.IsIPv6String(e2enode.GetAddresses(&node, v1.NodeInternalIP)[0]) {
-					newIP = "fc00:f853:ccd:e793:1111::" + strconv.Itoa(i)
-					nodeIPs[node.Name][6] = newIP
+
+				if isIPv6Cluster {
+					newIPIP := fmt.Sprintf("2001:db8:abcd:1234::%d", i+140)
+					nodeIPs[node.Name][6] = newIPIP
 				} else {
-					newIP = "172.18.1." + strconv.Itoa(i+1)
-					nodeIPs[node.Name][4] = newIP
+					newIPIP := fmt.Sprintf("10.10.10.%d", i+140)
```

Comment:
i think we used 172.18.1 to match the kind network subnet, does it matter that you are using something else here?

Reply:
its a new IP within the network 10.10.10.0/24:
```
secondaryIPV4Subnet                        = "10.10.10.0/24"
```
It was wrong before.